lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadata.php

```diff
@@ -454,6 +472,63 @@ public function validateIdentifier()
     }
 
     /**
+     * Validate that childClasses is empty if isLeaf is true.
+     *
+     * @throws MappingException if there is a conflict between isLeaf and childClasses.
+     */
+    public function validateChildClasses()
+    {
+        if (count($this->childClasses) > 0 && $this->isLeaf) {
+            throw new MappingException(sprintf(
+                'Cannot map a document as a leaf and define child classes for "%s"',
+                $this->name
+            ));
+        }
+    }
+
+    /**
+     * Assert that the given class FQN can be a child of the document this
+     * metadata represents.
+     *
+     * @param string $classFqn
+     * @throws OutOfBoundsException
+     */
+    public function assertValidChildClass(ClassMetadata $class)
+    {
+        $childClasses = $this->getChildClasses();
+
+        if ($this->isLeaf()) {
+            throw new OutOfBoundsException(sprintf(
+                'Document "%s" has been mapped as a leaf. It cannot have children',
+                $this->name
+            ));
+        }
+
+        if (array() === $childClasses) {
+            return;
+        }
+
+        $found = false;
+        foreach ($childClasses as $childClass) {
+            if ($class->name === $childClass || $class->reflClass->isSubclassOf($childClass)) {
+                $found = true;
```

Comment:
you can simply return here and abort the loop. and get rid of the $found variable, as you do nothing additional with it.